main/main.cpp

```diff
@@ -2530,9 +2530,9 @@ void Main::cleanup(bool p_force) {
 		memdelete(camera_server);
 	}
 
+	finalize_navigation_server();
 	OS::get_singleton()->finalize();
 	finalize_physics();
```

Comment:
Should physics be finalized before the OS singleton?

Reply:
No idea, in Godot 4 the entire chain is different but the OS also is finalized way later after physics.
Tbh I don't really want to touch it cause I don't know what is different and going on in Godot 3.